tests/test_blk/test_blk.c

```diff
@@ -59,18 +55,17 @@ int check_sector_write(uint64_t sector)
     return 0;
 }
 
-int solo5_app_main(char *cmdline __attribute__((unused)))
+int solo5_app_main(const char *cmdline __attribute__((unused)))
 {
     size_t i, nsectors;
-    int rlen;
 
     puts("\n**** Solo5 standalone test_blk ****\n\n");
 
     /*
      * Write and read/check one tenth of the disk.
      */
-    nsectors = solo5_blk_sectors();
-    for (i = 0; i <= nsectors; i += 10) {
+    nsectors = solo5_blk_capacity();
+    for (i = 0; i < nsectors; i += 10) {
```

Comment:
Yes, the sector offset is 0-based, so valid values are (0 ... solo5_blk_capacity() - 1). Interestingly there are "bugs" in the current backends (e.g. virtio/bhyve and ukvm) where writing past the end of the "device" would succeed if the underlying device was backed by a file which is the case in all of our tests.